src/fidesops/api/v1/endpoints/dataset_endpoints.py

```diff
@@ -166,35 +176,95 @@ def patch_datasets(
             "fides_key": dataset.fides_key,
             "dataset": dataset.dict(),
         }
-        try:
-            if connection_config.connection_type == ConnectionType.saas:
-                _validate_saas_dataset(connection_config, dataset)
-            # Try to find an existing DatasetConfig matching the given connection & key
-            dataset_config = DatasetConfig.create_or_update(db, data=data)
-            created_or_updated.append(dataset_config.dataset)
-        except (SaaSConfigNotFoundException, ValidationError) as exception:
-            logger.warning(exception.message)
-            failed.append(
-                BulkUpdateFailed(
-                    message=exception.message,
-                    data=data,
-                )
-            )
-        except Exception:
-            logger.warning(f"Create/update failed for dataset '{data['fides_key']}'.")
-            failed.append(
-                BulkUpdateFailed(
-                    message="Dataset create/update failed.",
-                    data=data,
-                )
-            )
+        create_or_update_dataset(
+            connection_config, created_or_updated, data, dataset, db, failed
+        )
+    return BulkPutDataset(
+        succeeded=created_or_updated,
+        failed=failed,
+    )
+
 
+@router.patch(
+    YAML_DATASETS,
+    dependencies=[Security(verify_oauth_client, scopes=[DATASET_CREATE_OR_UPDATE])],
+    status_code=200,
+    response_model=BulkPutDataset,
+    include_in_schema=False  # Not including this path in the schema.
+    # Since this yaml function needs to access the request, the open api spec will not be generated correctly.
+    # To include this path, extend open api: https://fastapi.tiangolo.com/advanced/extending-openapi/
+)
+async def patch_yaml_datasets(
+    request: Request,
+    db: Session = Depends(deps.get_db),
+    connection_config: ConnectionConfig = Depends(_get_connection_config),
+) -> BulkPutDataset:
+    if request.headers.get("content-type") != X_YAML:
+        raise HTTPException(
+            status_code=HTTP_415_UNSUPPORTED_MEDIA_TYPE,
+            detail="Supported type: " + X_YAML,
+        )
+    body = await request.body()
+    try:
+        yaml_request_body: dict = yaml.safe_load(body)
+    except yaml.YAMLError:
+        raise HTTPException(status_code=HTTP_400_BAD_REQUEST, detail="Error in YAML")
```

Comment:
Is there anything useful from `yaml.YAMLError` that we can include in the response to help fix any YAML errors?

Reply:
I pushed up a change to append the string version of the error.